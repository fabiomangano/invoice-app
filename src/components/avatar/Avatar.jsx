import React from 'react';
import PropTypes from 'prop-types';
import placeholderAvatar from '../../assets/image-avatar.jpg';
import './Avatar.style.scss';

const Avatar = ({ src, size, alt }) => (
  <img className="avatar" src={src} alt={alt} size={size} />
);

Avatar.defaultProps = {
  size: 'medium',
  src: placeholderAvatar,
};

Avatar.propTypes = {
  src: PropTypes.string,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  alt: PropTypes.string.isRequired,
};

export default Avatar;
