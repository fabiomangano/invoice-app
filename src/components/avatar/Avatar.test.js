import React from 'react';
import Renderer from 'react-test-renderer';
import Avatar from './Avatar';

describe('Avatar', () => {
  it('should render an avatar with a placeholder image', () => {
    let tree;
    Renderer.act(() => {
      tree = Renderer.create(<Avatar alt="image" />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
