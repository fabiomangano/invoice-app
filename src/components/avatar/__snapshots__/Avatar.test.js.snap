// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render an avatar with a placeholder image 1`] = `
<img
  alt="image"
  className="avatar"
  src="test-file-stub"
/>
`;
